core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java

```diff
@@ -269,12 +271,13 @@ public Void call() throws AlluxioException, IOException {
   }
 
   @Override
-  public void remove(final String path, final boolean recursive)
+  public void remove(final String path, final boolean recursive, final DeleteTOptions options)
       throws AlluxioTException, ThriftIOException {
     RpcUtils.call(new RpcCallableThrowsIOException<Void>() {
       @Override
       public Void call() throws AlluxioException, IOException {
-        mFileSystemMaster.delete(new AlluxioURI(path), recursive);
+        mFileSystemMaster.delete(new AlluxioURI(path), new DeleteOptions(options)
```

Comment:
for this part, could you reference how `free` is implemented. Basically we want to keep backward compatibility (so earlier Alluxio client could still talk to the latest Alluxio server). In your particular case,  earlier Alluxio client will not have options passed in (`options == null`)
```java

        if (options == null) {
          // For Alluxio client v1.4 or earlier.
          // NOTE, we try to be conservative here so early Alluxio clients will not be able to force
          // freeing pinned items but see the error thrown.
          mFileSystemMaster.delete(new AlluxioURI(path),
              DeleteOptions.defaults().setRecursive(recursive));
        } else {
          mFileSystemMaster.delete(new AlluxioURI(path), new DeleteOptions(options));
        }
        return null;
      
```